Removed Conditions trait from cluster. Improved label selectors.

maltesander opened the pull request:
## Description
- Removed conditions. 
- Sorted methods. 
- Bumped tokio to 1.10.
- Added list_config_maps. Renamed "get" to "list".

## Review Checklist
- [ ] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)

maltesander committed:
Removed conditions. Sorted methods. Bumped tokio to 1.10.
Added list_config_maps. Renamed "get" to "list".

maltesander committed:
Updated changelog.

soenkeliebau commented on src/operator/setup.rs:
```diff
@@ -184,6 +113,103 @@ where
+            .items
+    }
+
+    /// List all available pods in the cluster via the name label.
```
'Available' is misleading in this context I think. This refers to the existing pods that have been created in kube-apiserver, correct?

maltesander replied:
done

soenkeliebau commented on src/operator/setup.rs:
```diff
@@ -184,6 +113,103 @@ where
+    }
+
+    /// This creates a kube client and should be executed at the start of every test.
+    pub fn new(options: TestClusterOptions, timeouts: TestClusterTimeouts) -> Self {
```
nit: I would prefer this to be at the top of the impl block.

maltesander replied:
done

soenkeliebau reviewed (changes requested):
Left some comments

siegfriedweber commented on CHANGELOG.adoc:
```diff
@@ -4,11 +4,20 @@
 * Dependency `kube` set to version `0.58` ({15}).
 * Added operator checks for pod version (label) and pod creation timestamps ({19}).
 * Improved operator logging output (messages are prefixed with [<Kind>/<Name>]) ({19}).
+* Dependency `tokio` set to version `1.10` ({21}).
```
Today I discovered how Kubernetes lists its dependency changes: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.22.md#dependencies-1

It is just `<dependency>: <old version> → <new version>`.

maltesander replied:
I like that. Now we only provide the new version. Should we try that or are there any other proposals? @siegfriedweber @soenkeliebau @lfrancke @razvan

soenkeliebau replied:
I'd be happy to do it like that.

soenkeliebau replied:
Decided as group in the daily on 31.08.2021 to use Kubernetes notation.

maltesander replied:
done

maltesander committed:
Merge remote-tracking branch 'origin/main' into remove_conditions_trait

maltesander commented on CHANGELOG.adoc:
```diff
@@ -4,11 +4,20 @@
+
+=== Added
+* list_config_maps via label selector ({21}).
+* list_pods now has another label for cluster name ({21}).
```
done

maltesander commented on CHANGELOG.adoc:
```diff
@@ -4,11 +4,20 @@
 * Added operator checks for pod version (label) and pod creation timestamps ({19}).
 * Improved operator logging output (messages are prefixed with [<Kind>/<Name>]) ({19}).
+* Dependency `tokio` set to version `1.10` ({21}).
+* All "get_" methods renamed to "list_" ({21}).
```
done

maltesander commented on CHANGELOG.adoc:
```diff
@@ -4,11 +4,20 @@
+* Dependency `tokio` set to version `1.10` ({21}).
+* All "get_" methods renamed to "list_" ({21}).
+* Sorted cluster methods alphabetically ({21}).
+* Renamed `pod_name_label` to `cluster_type` ({21}).
```
done

maltesander commented on src/operator/setup.rs:
```diff
@@ -184,6 +113,103 @@ where
+            .items
+    }
+
+    /// Returns all available nodes in the cluster. Can be used to determine the expected pods
```
I improved it

maltesander committed:
Adapted to PR review.

soenkeliebau commented:
Just that one comment about notation for dependency updates left to do, after that happy to approve.

maltesander committed:
Adapted dependencies retroactive to the agreed upon format (<dependency>: <old version> → <new version>).
Moved Breaking changes to the top. Added Breaking [Operator/Agent] indicator if changes only affect one of the two.

soenkeliebau reviewed (approved):
LGTM
